src/lib/externaldata.py

```diff
@@ -248,6 +248,23 @@ def from_source(
         if not source.get("url"):
             raise SourceUnsupported('Data is not external: no "url" property')
 
+        if source.get("url"):
+
+            if (
+                source.get("type") == ExternalBase.Type.EXTRA_DATA
+                and "sha256" not in source.keys()
+            ):
+                raise SourceLoadError("Extra data is missing SHA256 checksum")
+
+            if source.get("type") in (
+                ExternalBase.Type.FILE,
+                ExternalBase.Type.ARCHIVE,
+            ):
+                if not any(
+                    k in source.keys() for k in ("md5", "sha1", "sha256", "sha512")
+                ):
+                    raise SourceLoadError("Data is missing checksum")
+
```

Comment:
The `if source.get("url")` guard is unneeded because it was checked right above.

You can also simplify the checksum checks:

```suggestion
        if (
            source.get("type") == ExternalBase.Type.EXTRA_DATA
            and "sha256" not in source
        ):
            raise SourceLoadError("Extra data is missing SHA256 checksum")

        if source.get("type") in (
            ExternalBase.Type.FILE,
            ExternalBase.Type.ARCHIVE,
        ):
            if not (source.keys() & {"md5", "sha1", "sha256", "sha512"}):
                raise SourceLoadError("Data is missing checksum")
```

Reply:
I don't want it to trigger when path was used, that does not require a checksum.

Reply:
The check above:

```python3
        if not source.get("url"):
            raise SourceUnsupported('Data is not external: no "url" property')
```

guarantees that `if source.get("url")` is true if reached, so the additional check is unnecessary. Or am I missing something?

Reply:
Ah no you're right, I didn't see the line above.